Shared.Tests/CloudinaryTest.cs

```diff
@@ -2691,70 +2758,66 @@ public void SearchResourceByTag()
         [Test]
         public void SearchResourceByPublicId()
         {
-            string publicId = string.Concat(m_suffix, "_TestForTagSearch");
+            string publicId = $"{m_apiTest}_SearchResourceByPublicId";
 
             var uploadParams = new ImageUploadParams()
             {
                 File = new FileDescription(m_testImagePath),
-                
                 PublicId = publicId,
                 Overwrite = true,
-                Type = "private"
+                Type = "private",
+                Tags = m_apiTag
             };
-            var uploadResult = m_cloudinary.Upload(uploadParams);
+            m_cloudinary.Upload(uploadParams);
             Thread.Sleep(10000);
             var result = m_cloudinary.Search().Expression(string.Format("public_id: {0}", publicId)).Execute();
             Assert.True(result.TotalCount > 0);
-            DelResResult delResult = m_cloudinary.DeleteResources(new string[] { publicId });
         }
 
         [Test]
         public void TestSearchResourceByExpression()
         {
-            string publicId = string.Concat(m_suffix, "_TestForSearch");
+            string publicId = $"{m_apiTest}_TestSearchResourceByExpression";
```

Comment:
We might want to make a helper function that would call:
System.Reflection.MethodBase.GetCurrentMethod().Name and populate public ID based on it